Jenkinsfile

```diff
@@ -100,6 +101,23 @@ def init_git() {
     script: './tests/scripts/task_show_node_info.sh',
     label: 'Show executor node info',
   )
+
+  // Determine merge commit to use for all stages
+  sh(
+    script: 'git fetch origin main',
+    label: 'Fetch upstream',
+  )
+  if (upstream_revision == null) {
```

Comment:
q: what about race conditions with this approach?

Reply:
Right now it's ok since this should happen during the `Sanity Check` stage which gates parallel execution on a sequential job